.gitlab-ci.yml

```diff
@@ -313,6 +313,7 @@ build_dogstatsd_static-deb_x64:
   stage: binary_build
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: [ "runner:main", "size:large" ]
+  needs: [ "run_dep_check_lock" ]
```

Comment:
So, IMO, all of the stuff in the `binary build` stage (now that you've moved the CF windows which takes a while) should depend on the entire test stage (or at least the relevant test).
These builds are short enough that there's no reason to give them a head start, and we can skip them altogether if the build fails.

Reply:
👍 I'm going to remove this commit.

Reply:
About mixing both DAG and stages, I think in some cases it makes some sense. Eg: for the `deploy` stages, I like the idea that they only run if everything before they is green (without having to add lots of `needs`).